js/modules/k6/ws/ws.go

```diff
@@ -262,9 +262,9 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 		case readErr := <-readErrChan:
 			socket.handleEvent("error", rt.ToValue(readErr))
 
-		case readClose := <-readCloseChan:
```

Comment:
Okay, so correct me if I am wrong, but: 
socket.Close is going to call the `handleEvent("close", ...)` but only with the `GoingAwayClose` code, which might not be the code that it received from the remote end and this is the handling of the remote end.... 
IMO this is clearly wrong. Reading the [relevant RFC](https://tools.ietf.org/html/rfc6455#page-36) I think that what we should do is:
1. handleEvent close with the code that was returned by the server. 
2. send close frame with the same code to the server. 
Here the question is, whether we also handle the second closer as well ? I think that we shouldn't.

Reply:
I considered this, but ultimately left it as is since it's not technically against the RFC:

> (When sending a Close frame in response, the endpoint typically echos the status code it received.)

This is poorly defined, but I would interpret it as "MAY"/"SHOULD", so opted against being fully compliant for now, and fixing this specific issue in this PR.

Would it make sense to open an issue and address WS compatibility in another PR?

Reply:
I added a `readPump()` test in 0f96be17.

This doesn't test the entire flow of k6 sending a close with the same code, but that would involve a more complicated `WS.Connect()` test I'd rather leave after the WS refactor when this is hopefully easier to test.